pkg/api/config.go

```diff
@@ -103,3 +104,53 @@ func (config *ReleaseBuildConfiguration) DependencyParts(dependency StepDependen
 	}
 	return stream, name, explicit
 }
+
+// WithPresubmitFrom returns a new configuration, where a selected test from the source
```

Comment:
It might almost make sense to think of it the other way - what do we want the root config to provide? Is it *just* the images? It's potentially hard to know "everything" that a test may need from the source config, could it be easier to just take source config, add images in, and call it a day? At least that is pretty bounded (base_images, images, raw_steps?)

Reply:
I think that's a good point. I think that it's easier to reason about the API in the direction I wrote it here (inject a test into a config), and the implementation may potentially differ if we find some use for it. But I gave it a mental experiment:

I think that transitive dependencies of a test (images, releases) are ultimately similar to transitive dependencies of images (images, releases), so you don't really save any work or complexity. I think that the source (root in your sentence) config mainly provides everything needed to build the images from source, so you need `images`, images likely need `src` (=`binary_build_command`), src needs build_root, you potentially need `operators` (although operators testing are beyong what I attempt to do here). You need base_images union. Maybe raw steps, but likely not.

I don't think anything like this will ever work (without a significant change in ci-operator) if the injected test depends on an image that's actually built in the source config - that implies you need access to the source code of source config's repo, and you don't have that in base config's repo presubmit.

So I'm leaning towards keeping it this way.